packages/common/src/writeContract.ts

```diff
@@ -63,43 +71,59 @@ export async function writeContract<
     queueConcurrency: opts.queueConcurrency,
   });
 
-  async function prepareWrite(): Promise<
-    WriteContractParameters<abi, functionName, args, chain, account, chainOverride>
-  > {
+  const feeRef = getFeeRef({ client: opts.publicClient ?? client, refreshInterval: 10000 });
+
+  async function prepare(): Promise<WriteContractParameters<abi, functionName, args, chain, account, chainOverride>> {
     if (request.gas) {
-      debug("gas provided, skipping simulate", request.functionName, request.address);
+      debug("gas provided, skipping preparation", request.functionName, request.address);
       return request;
     }
 
-    debug("simulating", request.functionName, "at", request.address);
-    const result = await simulateContract<chain, account | undefined, abi, functionName, args, chainOverride>(
```

Comment:
we're no longer simulating?

iirc there was some nuance between estimating gas vs simulating where estimating gas has less nice errors than simulating, and maybe even didn't surface certain kinds of errors ahead of time?

Reply:
discussed offline: viem is just doing gas estimation internally, and simulate doesn't give you a gas estimate, so it'd be one more rpc call. Using gas estimate for now and can do an additional simulate in case of a revert later if we find that gas estimation doesn't give good errors.

Reply:
to put another way: we're gonna follow viem defaults here and hope for the best 🤞 

we can come back and add simulation again if we find we get better error handling and less reverts